client/core/trade.go

```diff
@@ -728,10 +755,13 @@ func (t *trackedTrade) swapMatches(matches []*matchTracker) error {
 	log.Infof("Broadcasted transaction with %d swap contracts for order %v. Fee rate = %d. Receipts: %v",
 		len(receipts), t.ID(), swaps.FeeRate, receipts)
 
-	t.change = change
-	t.coins[hex.EncodeToString(change.ID())] = change
-	t.metaData.ChangeCoin = []byte(change.ID())
-	t.db.SetChangeCoin(t.ID(), t.metaData.ChangeCoin) // TODO: lock the change coin in the wallet if the order is still on the book
+	if change != nil {
+		log.Debugf("storing change coin %v", change.String())
+		t.change = change
+		t.coins[hex.EncodeToString(change.ID())] = change
+		t.metaData.ChangeCoin = []byte(change.ID())
+		t.db.SetChangeCoin(t.ID(), t.metaData.ChangeCoin) // TODO: lock the change coin in the wallet if the order is still on the book
+	}
```

Comment:
I think we'd still want to save the change coin if it's `nil`, no? We don't want to append it to `t.coins` though.

Reply:
Hmm, yeah `t.change = nil ` would still be desirable, but `change.ID()` would of course panic with `t.metaData.ChangeCoin = []byte(change.ID())`.  Probably OK to handle this with an `else` with `t.metaData.ChangeCoin = nil`.

BTW, this gets to the other comment about `ChangeCoin: oBkt.Get(changeKey)` where I'm not certain that `oBkt.Put(nil)` comes back as nil.  Just need to check that.